### ISSUE TITLE
vg construct maf import: alignment block parsing

### ISSUE THREAD

**Issue by ChriKub** [vgteam/vg#1577]
Hi @jeizenga ,

I'm trying to load a progressiveCactus alignment into vg. The loading and conversion into a vg graph works just fine, but the final vg graph does not correctly represent the alignment in the given maf file. All alignment blocks are represented as separate paths instead of concatenating the alignment blocks in the correct order to represent the full path.

vg explode on the imported alignment returns me a component for every alignment block in the maf: 

> ./component0.vg        9583.G333.1     9721.G310.1     9792.G422.1     Anc0.Anc0refChr0.1
> ./component1.vg        9583.G333.2     9721.G310.2     9792.G422.2     9911.G415.1     Anc0.Anc0refChr0.2
> ./component2.vg        9583.G333.3     9721.G310.3     9792.G422.3     9911.G415.2     Anc0.Anc0refChr0.3
> ./component3.vg        9583.G333.4     9721.G310.4     9792.G422.4     9911.G415.3     Anc0.Anc0refChr0.4
> ./component4.vg        9583.G333.5     9721.G310.5     9792.G422.5     9911.G415.4     Anc0.Anc0refChr0.5
> ./component5.vg        9583.G333.6     9721.G310.6     9792.G422.6     9911.G415.5     Anc0.Anc0refChr0.6
> 

instead of a single component representing the alignment:

>./component0.vg        9583.G333     9721.G310     9792.G422     Anc0.Anc0refChr0

Data to recreate the issue can be downloaded [here](https://owncloud.tuebingen.mpg.de/index.php/s/op9iZeJlkag5LbY)

```
vg construct -M 15.2.maf -F maf > 15.2.vg
vg explode 15.2.vg ./
```

Thanks,
Chris

Please describe:

1. What you were trying to do
2. What you wanted to happen
3. What actually happened
4. What data and command line to use to make the problem recur, if applicable

```
Format code blocks or terminal copy-pastes like this, between triple backticks.
```

**Comment by fbemm** [comment]
I disagree with both comments. MAF is designed to particularity coop with multiple pieces of the/an alignment independent if the break is introduced by a simple line width limit or an sub-alignment boundary. Especially progressiveCactus, Mauve and other tools (practically all tools naturally outputting MAF format) have a simple hard limit when to break an alignment block. This has nothing to do with the alignment itself but just with its readability. The same is true for  the clustal format, which suffers from the same parsing problems. Alignments are usually split up into different blocks not because the alignment breaks but just to keep the format readable on the CLI.

**Comment by jeizenga** [comment]
Okay, I've talked to some people over here who are more familiar with the MAF format than I am, and it looks like it's even more complicated that you're describing. The MAF format can describe fully general graph-theoretic alignments (even including self-alignments). Accordingly, it's going to take some heavier-duty algorithm development to support it correctly (probably using transitive alignment primitives from https://github.com/benedictpaten/pinchesAndCacti). I'll get around to this, but I can't promise it will get done particularly soon.
